### PR TITLE
Add loss normalization to normalize_intensities filter

### PR DESCRIPTION
**Pull request by florian-huber**
- Add loss normalization to ``normalize_intensities`` filter
- Add case to unit tests.
- Slightly simplify ``add_losses`` function

### CODE REVIEW

**Comment by sverhoeven** [comment]
Blocks https://github.com/iomega/spec2vec/pull/12

File: matchms/filtering/add_losses.py [hunk 1]
```diff
@@ -15,22 +15,15 @@ def add_losses(spectrum_in: SpectrumType, loss_mz_from=0.0, loss_mz_to=1000.0) -
...
     if spectrum_in is None:
         return None
 
     spectrum = spectrum_in.clone()
 
     precursor_mz = spectrum.get("precursor_mz")
     if precursor_mz:
-        assert precursor_mz_is_number(), ("Expected 'precursor_mz' to be a scalar number.",
-                                          "Consider applying 'add_precursor_mz' filter first.")
+        assert isinstance(precursor_mz, (float, int)), ("Expected 'precursor_mz' to be a scalar number.",
```
Comment (sverhoeven) [inline matchms/filtering/add_losses.py:25]: Nice, much cleaner then before

**Review by sverhoeven — APPROVED** [review]

File: matchms/filtering/add_losses.py (hunk 1 above)
Comment (florian-huber) [reply matchms/filtering/add_losses.py:25]: Thanks.

**Comment by florian-huber** [comment]
Thanks a lot for the review Stefan!

**Pull request merged**